### ISSUE TITLE
Widgets with Mac-native scrollbars (eg wxDataViewCtrl) do not pass scroll state to wxWindow or override GetScrollPosition etc.

### ISSUE THREAD

**Issue by danhd123** [wxWidgets/wxWidgets#24137]
#### Bug description:

`int GetScrollPosition(int orientation)` and related methods on `wxWindow` subclasses, such as `wxDataViewCtrl`, that have Mac-native controls with builtin scrollbars, do not work. This makes it impossible to, eg. save and restore scroll position in a `wxDataViewCtrl`.

#### Expected vs observed behaviour:

I expect  SetScrollbar, SetScrollPos, GetScrollPos, GetScrollThumb, GetScrollRange, ScrollWindow, ScrollLines, ScrollPages, AlwaysShowScrollbars, and IsScrollbarAlwaysShown to work in wxDataViewCtrl, its subclasses, and possibly other classes, that have native built-in scrollbars, to work, and `CanScroll` to return `true.

`CanScroll` incorrectly returns `false`, the rest do nothing or return 0 values.

#### Patch or snippet allowing to reproduce the problem:

```
class MyCtrl : public wxDataViewCtrl {
  MyCtrl(wxWindow* pParent)
    : wxDataViewCtrl(pParent, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxDV_MULTIPLE | wxBORDER_NONE | wxDV_ROW_LINES) {
    std::cout << CanScroll();
    }
}; 
```

This should print true. It prints false. Try adding items, scrolling, and getting the scroll position. It will always be 0. Try setting the scroll position. It does nothing.

#### To Reproduce:

1. Look at https://github.com/wxWidgets/wxWidgets/blob/6d89a14c7256e9d4723f3b212a760ce0152fe7e5/src/osx/dataview_osx.cpp#L445. Note that the flags for wxWidgets scrollbars are removed to prevent a crash
2. Look at https://github.com/wxWidgets/wxWidgets/blob/6d89a14c7256e9d4723f3b212a760ce0152fe7e5/src/common/wincmn.cpp#L1125.
3. Look at https://github.com/wxWidgets/wxWidgets/blob/6d89a14c7256e9d4723f3b212a760ce0152fe7e5/include/wx/osx/dataview.h. Note that there is no override of any scrolling-related method.

### Platform and version information

- wxWidgets version  you use: 3.2.4
- wxWidgets port  you use: wxOSX
- OS  and its version: macOS 13.6.1

**Comment by vadz** [comment]
I don't think this problem is limited to wxOSX, unfortunately. wx scrolling methods typically work with scrolling implemented in wx but not with the native controls, so I think the best we can do is to document it -- unless someone is ready to actually make this work, of course.
